### PR TITLE
Stratospheric harness fix - issue #79

### PR DESCRIPTION
**Pull request by PapaGuacamole**
Had two forks with similar names, updates were being pushed the wrong way.

Moved creation / removal of harness pathing for Achaea to separate functions
Uses different area IDs for Targossas dependent on mmp.settings.crowdmap (relates to PR78)

### CODE REVIEW

File: mudlet-mapper.xml [hunk 1]
```diff
@@ -3296,6 +3296,47 @@ end</script>
...
+            <Trigger isActive="yes" isFolder="no" isTempTrigger="no" isMultiline="yes" isPerlSlashGOption="no" isColorizerTrigger="no" isFilterTrigger="no" isSoundTrigger="no" isColorTrigger="no" isColorTriggerFg="no" isColorTriggerBg="no">
+                <name>City has orb of confinement</name>
+                <script>local cityIDs = {
+	49,		-- Ashtan
+	57,		-- Cyrene
+	51,		-- Eleusis
+	55,		-- Hashan
+	44,		-- Mhaldor
+	368,	-- Targossas
+}
```
Comment (keneanung) [inline mudlet-mapper.xml]: Do you need this table here? I can't see it being used.

File: mudlet-mapper.xml [hunk 2]
```diff
@@ -5377,17 +5426,29 @@ function speedwalking(event, num)
...
         table.contains(gmcp.Room.Info.details, &quot;indoors&quot;) or
         table.contains(gmcp.Room.Info.details, &quot;considered indoors&quot;)
-      )
+      ) and
+			not (getRoomArea(mmp.currentroom) == 49 and mmp.settings.orbashtan) and
+			not (getRoomArea(mmp.currentroom) == 57 and mmp.settings.orbcyrene) and
+      not (getRoomArea(mmp.currentroom) == 51 and mmp.settings.orbeleusis) and
+      not (getRoomArea(mmp.currentroom) == 55 and mmp.settings.orbhashan) and
+      not (getRoomArea(mmp.currentroom) == 44 and mmp.settings.orbmhaldor) and
+      not (getRoomArea(mmp.currentroom) ==368 and mmp.settings.orbtargossas)
```
Comment (keneanung) [inline mudlet-mapper.xml]: You'll probably want to use the `orbed` function here as this area ID is crowdmap only.

File: mudlet-mapper.xml [hunk 3]
```diff
@@ -5377,17 +5426,29 @@ function speedwalking(event, num)
...
-        table.contains(gmcp.Room.Info.details, &quot;considered indoors&quot;)
-      )
-    then
+        table.contains(gmcp.Room.Info.details, &quot;considered indoors&quot;) or
+        (getRoomArea(mmp.currentroom) == 49 and mmp.settings.orbashtan) or
+        (getRoomArea(mmp.currentroom) == 57 and mmp.settings.orbcyrene) or
+        (getRoomArea(mmp.currentroom) == 51 and mmp.settings.orbeleusis) or
+        (getRoomArea(mmp.currentroom) == 55 and mmp.settings.orbhashan) or
+        (getRoomArea(mmp.currentroom) == 44 and mmp.settings.orbmhaldor) or
+        (getRoomArea(mmp.currentroom) ==368 and mmp.settings.orbtargossas)
```
Comment (keneanung) [inline mudlet-mapper.xml]: This should probably be `orbed` as well.

File: mudlet-mapper.xml [hunk 4]
```diff
@@ -9153,6 +9175,197 @@ registerAnonymousEventHandler(&quot;mmapper arrived&quot;, &quot;mmp.WearShackle
...
+    not (mmp.game == &quot;achaea&quot;) or
+    (mmp.currentroom == mmp.speedWalkPath[#mmp.speedWalkPath]) or
+    table.contains(gmcp.Room.Info.details, &quot;indoors&quot;) or
+    (
+      not (getRoomArea(mmp.currentroom) == 49 and mmp.settings.orbashtan) and
+      not (getRoomArea(mmp.currentroom) == 57 and mmp.settings.orbcyrene) and
+      not (getRoomArea(mmp.currentroom) == 51 and mmp.settings.orbeleusis) and
+      not (getRoomArea(mmp.currentroom) == 55 and mmp.settings.orbhashan) and
+      not (getRoomArea(mmp.currentroom) == 44 and mmp.settings.orbmhaldor) and
+      not (getRoomArea(mmp.currentroom) == 368 and mmp.settings.orbtargossas)
```
Comment (keneanung) [inline mudlet-mapper.xml]: And another place for the `orbed` function.

File: mudlet-mapper.xml [hunk 5]
```diff
@@ -9153,6 +9175,197 @@ registerAnonymousEventHandler(&quot;mmapper arrived&quot;, &quot;mmp.WearShackle
...
+      function()
+        if mmp.settings.orbtargossas then
+          mmp.echo(&quot;Orb of Confinement set as &lt;green&gt;ACTIVE&lt;reset&gt; for Targossas&quot;)
+        else
+          mmp.echo(&quot;Orb of Confinement set as &lt;red&gt;INACTIVE&lt;reset&gt; for Targossas&quot;)
+        end
+      end,
+  }
+
+function mmp.setOrb(areaID)
```
Comment (keneanung) [inline mudlet-mapper.xml]: I'd probably rewrite this function like this:
```lua
function mmp.setOrb(city)
  mmp.clearpathcache()
  if mmp.settings["orb"..city:lower()] then
    mmp.echo("Orb of Confinement set as <green>ACTIVE<reset> for " .. city:title())
  else
    mmp.echo("Orb of Confinement set as <red>INACTIVE<reset> for " .. city:title())
  end
end
```

and have the lambda in the `createOption` call be
```lua
function() mmp.setOrb("targossas") end
```

I think that makes the call cleaner and gets rid of the `cityIDs` table which includes a lot of duplicated code.

Comment (PapaGuacamole) [reply mudlet-mapper.xml]: Made the changes as requested

**Pull request merged**